### PR TITLE
Nhsn indicator

### PR DESCRIPTION
**Pull request by aysim319**
### Description
Adding new indicator NHSN

### Associated Issue(s) 
- Addresses #2079

### CODE REVIEW

**Comment by nmdefries** [comment]
Due to production issues, will aim to release this early next week.

**Comment by aysim319** [comment]
context from david:
> oh, also, dunno if you saw the post from logan, they're doing this really dumb thing of putting the preliminary data in a separate source https://data.cdc.gov/Public-Health-Surveillance/Weekly-Hospital-Respiratory-Data-HRD-Metrics-by-Ju/mpgq-jmmr/about_data
> we should probably include those as separate versions in the same indicator
> oh er, when I said versions, I meant versions (like the release date) rather than different signals. Definitely not the best wording choice.

Currently have it such that they're seperate signals similar to nssp secondary source to differienate them.

**Comment by dsweber2** [comment]
Working through this. I tried a couple of smoothings (3 & 4 weeks, both trailing and centered) to see if that would help tame some of the more extreme data issues, but it didn't really help. I had a use-case where hhs_regions would've been handy, so that's probably worth adding eventually (probably more important to get this live first though).

File: nhsn/tests/test_pull.py
Comment (dsweber2) [inline nhsn/tests/test_pull.py:2]: apparently unused
```suggestion

```

File: nhsn/tests/test_run.py
Comment (dsweber2) [inline nhsn/tests/test_run.py:11]: I'm sure this comment made sense at one point >.<

File: nhsn/tests/test_run.py
Comment (dsweber2) [inline nhsn/tests/test_run.py:1]: I'm a little confused how the files that this is checking for get defined. Are they originating with `test_pull`?

**Review by dsweber2 — APPROVED** [review]
Looks good to me! A few things I'm a bit confused about, and some minor notes, but I would be glad to see what this does on staging.

File: nhsn/delphi_nhsn/pull.py
Comment (minhkhul) [reply nhsn/delphi_nhsn/pull.py:55]: I believe this is holdover from my nssp and nchs dirty code practice. If we aren't doing a full rewrite soon, I would refactor them at some point. For now, they all work (ugly).

File: nhsn/tests/test_run.py
Comment (aysim319) [reply nhsn/tests/test_run.py:1]: in line 21, one of the parameters for the test is `run_as_module` which is a fixture that was defined in conftest.
fixtures are pytest helpers that prepares everything for your test and makes sure it's cleaned up afterward.

File: nhsn/delphi_nhsn/constants.py
Comment (aysim319) [reply nhsn/delphi_nhsn/constants.py:16]: nope, the create_export_csv specifically looks for `timestamp` I believe acquisition parses the information to show up as `time_value`

File: nhsn/DETAILS.md
Comment (aysim319) [reply nhsn/DETAILS.md:12]: release got pushed back from environment issues, so I can add hhs...depending on the other checklist for this pr, might not be able to make it fyi

File: nhsn/delphi_nhsn/pull.py
Comment (aysim319) [reply nhsn/delphi_nhsn/pull.py:55]: I think I was bouncing back and forth, part of it being what if the other data source has different transformations down the line? maybe better to have a method defined? something along those lines. I did refactor the pull_data, since that doesn't seem like it would change

File: nhsn/DETAILS.md
Comment (dsweber2) [reply nhsn/DETAILS.md:12]: either way, we've got plenty of things to try in forecasting that don't depend on it

File: nhsn/params.json.template
Comment (minhkhul) [inline nhsn/params.json.template:4]:
```suggestion
    "log_filename": "nhsn.log"
```

File: nhsn/params.json.template
Comment (minhkhul) [inline nhsn/params.json.template:8]:
```suggestion
      "data_source": "nhsn",
```

File: nhsn/params.json.template
Comment (minhkhul) [inline nhsn/params.json.template:13]: Check on these [here](https://github.com/cmu-delphi/covidcast-indicators/blob/feea58d900cdb2ac4988d4cf1fb9ef96bbb54181/_delphi_utils_python/delphi_utils/validator/README.md)

Comment (aysim319) [reply nhsn/params.json.template:13]: ?

Comment (minhkhul) [reply nhsn/params.json.template:13]: Validation is triggered in production runs when there's a validation section in params.json like this and the runner is used to trigger the indicator run (`env/bin/python -m delphi_utils.runner delphi_${ind_name}` instead of `env/bin/python -m delphi_${ind_name}`).

You can remove the whole validation section if it's not needed for the indicator. But if you want to run the indicator with validation, the link has some more info on how to configure it in accordance with lag behaviors and such.

**Review by minhkhul — CHANGES_REQUESTED** [review]

File: nhsn/params.json.template
Comment (minhkhul) [inline nhsn/params.json.template:4]:
```suggestion
    "log_filename": "nhsn.log",
```

File: nhsn/params.json.template
Comment (minhkhul) [inline nhsn/params.json.template:5]:
```suggestion
    "log_filename": "nhsn.log",
    "backup_dir": "./raw_data_backups"
```

**Review by minhkhul — APPROVED** [review]

**Comment by nmdefries** [comment]
[Plots for statistical review](https://github.com/user-attachments/files/18043510/plots.zip) have been approved

**Comment by aysim319** [comment]
Waiting for signal name approval

**Comment by aysim319** [comment]
> Waiting for signal name approval

new signal name approved

**Pull request merged**
